src/common/mfu_io.c

```diff
@@ -2101,14 +2060,53 @@ DIR* daos_opendir(const char* dir, mfu_file_t* mfu_file)
         errno = ENOMEM;
         return NULL;
     }
-    int rc = dfs_lookup(mfu_file->dfs, dir, O_RDWR, &dirp->dir, NULL, NULL);
-    if (dirp->dir == NULL) {
-        MFU_LOG(MFU_LOG_ERR, "dfs_lookup %s failed", dir);
-        errno = rc;
-        free(dirp);
-        return NULL;
+
+    char* name     = NULL;
+    char* dir_name = NULL;
+    parse_filename(dir, &name, &dir_name);
+    assert(dir_name);
+
+    if (!name || strcmp(name, "/") == 0) {
+        /* For root, just lookup the entry */
+        int rc = dfs_lookup(mfu_file->dfs, dir, O_RDWR, &dirp->dir, NULL, NULL);
+        if (rc) {
+            MFU_LOG(MFU_LOG_ERR, "dfs_lookup %s failed", dir);
+            errno = rc;
+            free(dirp);
+            return NULL;
+        }
+    } else {
+        /* for non-root, try to cache the parent */
+        dfs_obj_t* parent = daos_hash_lookup(dir_name, mfu_file);
+        if (parent == NULL) {
+            goto err_dirp;
+        } else {
+            mode_t mode;
+            int rc = dfs_lookup_rel(mfu_file->dfs, parent, name, O_RDWR, &dirp->dir,
+                                   &mode, NULL);
+            if (rc) {
+                MFU_LOG(MFU_LOG_ERR, "dfs_lookup_rel %s/%s failed", dir_name, name);
+                errno = rc;
+    	        goto err_dirp;
```

Comment:
spacing?

Reply:
👍 Another case of tabs instead of spaces. Thanks!